### PR TITLE
StuPoli Production environment with Nginx reverse proxy

### PR DESCRIPTION
**Pull request by sebastianapw**
I've added local and production environment with docker-compose for use at the StuPoli Hamburg. I also prepared a settings and env file for the clinic.

### CODE REVIEW

File: config/nginx/production/templates/default.conf [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+server {
+    listen 443 ssl;
+    server_name localhost;
+
+    ssl_certificate           /etc/nginx/certs/cert.crt;
+    ssl_certificate_key       /etc/nginx/certs/cert.key;
+
+    ssl_session_cache  builtin:1000  shared:SSL:10m;
+    ssl_protocols  TLSv1 TLSv1.1 TLSv1.2;
```
Comment (bjubes) [inline config/nginx/production/templates/default.conf:23]: Should only allow TLSv1.2 and TLSv1.3. all other versions are considered insecure.

File: .envs/.production/.stupoli [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# General
+# ------------------------------------------------------------------------------
+DJANGO_SETTINGS_MODULE=config.settings.production
+
+DJANGO_SECRET_KEY='uF0sHMCxlAmPJlxpIY8QLUJElAWigxVdOwC92qe0l7Q7LBMfXNWSWQWxV34D8x53'
```
Comment (bjubes) [inline .envs/.production/.stupoli:5]: this value needs to be in a .secrets file that is gitignored so it does not appear in git.

File: .envs/.production/.stupoli [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+DJANGO_SERVER_EMAIL=roboter@stupoli-hamburg.de
+DJANGO_ADMIN_URL=admin 
+
+# PostgreSQL
+# ------------------------------------------------------------------------------
+POSTGRES_HOST=postgres
+POSTGRES_PORT=5432
+POSTGRES_DB=osler
+POSTGRES_USER=stupoli
+POSTGRES_PASSWORD=stupoli
```
Comment (bjubes) [inline .envs/.production/.stupoli:27]: this also needs to be in a secrets file

File: config/settings/stupoli.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+OSLER_DISPLAY_FOLLOWUP = False
+OSLER_DISPLAY_VACCINE = False
+
+OSLER_DEFAULT_CITY = "Hamburg"
+OSLER_DEFAULT_STATE = "Hamburg"
+OSLER_DEFAULT_ZIP_CODE = "20459"
+OSLER_DEFAULT_COUNTRY = "Deutschland"
+OSLER_DEFAULT_ADDRESS = "Seewartenstraße 10"
+
+OSLER_ABOUT_NAME = "wiki.stupoli-hamburg.de"
```
Comment (wwick) [inline config/settings/stupoli.py:26]: The about name should just be a name, not the actual url.

You should add an OSLER_ABOUT_URL setting, making its default the GitHub url, then overriding to your wikipedia page for StuPoli and the sojourner site that was previously linked for UMKC.

**Review by wwick — CHANGES_REQUESTED** [review]
Make absolutely sure that any of the private credentials which were exposed in a previous commit are changed because they will live on the internet forever.

Additionally, when I tried to run `local-stupoli.yml` on my computer, I got an error saying "Firefox can’t establish a connection to the server at 0.0.0.0:8000." and similar errors with other browsers. I'm not sure what the actual problem is, but it seems that local settings do not work at the moment.

File: config/settings/stupoli.py (hunk 4 above)
Comment (sebastianapw) [reply config/settings/stupoli.py]: I've added a OSLER_ABOUT_URL and set it to the GitHub url in base.py, thanks for the hint!

File: config/settings/base.py [hunk 5]
```diff
@@ -377,8 +377,12 @@
     "OSLER_ATTENDANCE_URL",
     default="https://www.wustl.edu",
 )
+OSLER_ABOUT_URL = env(
```
Comment (wwick) [inline config/settings/base.py:380]: Having this as an environmental variable which can also be overridden seems redundant, but that's not a big issue

**Review by wwick — CHANGES_REQUESTED** [review]
I'm still getting "Can't connect to server" when I try to run local-stupoli.yml

**Pull request merged**